### ISSUE TITLE
`Node.assertDescendant` could not find node - when copy/pasting nested list fragment

### ISSUE THREAD

**Issue by jankurianski** [ianstormtaylor/slate#3081]
#### Do you want to request a _feature_ or report a _bug_?

Report a bug.

#### What's the current behavior?

Code base to reproduce: https://codesandbox.io/s/slatejs-copypaste-nested-list-crash-gxphv

Steps to reproduce:

![GIF of reproduction](https://i.imgur.com/0Ug5xkR.gif)

1. Go to https://gxphv.csb.app
2. Place the cursor in front of the character B.
3. Hold down Shift and press the Down arrow key.
4. Copy (CTRL+C on Windows)
5. Place the cursor after the character D.
6. Paste (CTRL+V on Windows)
7. Nothing occurs.  In the JavaScript console, an error is reported:

```
Uncaught Error: `Node.assertDescendant` could not find node with path or key: 34
    at Block.ElementInterface.<computed> [as assertDescendant] (slate.js:15764)
    at Block.removeNode (slate.js:14795)
    at eval (slate.js:6271)
    at Editor.withoutNormalizing (slate.js:11221)
    at Editor.withoutNormalizing (slate-react.js:5948)
    at Commands$1.insertFragmentAtRange (slate.js:6210)
    at Editor.command (slate.js:10930)
    at Editor.command (slate-react.js:5892)
    at onCommand (slate.js:5225)
    at next (slate.js:11131)
    at onCommand (slate.js:5224)
    at next (slate.js:11131)
    at onCommand (slate.js:5224)
    at next (slate.js:11131)
    at Editor.run (slate.js:11165)
    at Editor.command (slate.js:10937)
    at Editor.method [as insertFragmentAtRange] (slate.js:11058)
    at Commands$6.insertFragment (slate.js:10266)
    at Editor.command (slate.js:10930)
    at Editor.command (slate-react.js:5892)
    at onCommand (slate.js:5225)
    at next (slate.js:11131)
    at onCommand (slate.js:5224)
    at next (slate.js:11131)
    at onCommand (slate.js:5224)
    at next (slate.js:11131)
    at Editor.run (slate.js:11165)
    at Editor.command (slate.js:10937)
    at Editor.method [as insertFragment] (slate.js:11058)
    at onPaste (slate-react.js:4677)
    at next (slate.js:11131)
    at onPaste (slate-react.js:5166)
```

This bug occurs with Slate 0.47.8, Slate React 0.22.9.  Chrome 77, Firefox 70, Edge 44 and Safari 12.1.

#### What's the expected behavior?

Paste works.

A workaround for now is to create an `onPaste` plugin that first tries to use `ed.insertFragment(getEventTransfer(event).fragment)` and then in case of an error, falls back to  `ed.insertFragment(new Html().deserialize(getEventTransfer(event).html))` as the html is also provided in case of a paste event of type `fragment`.

I've noticed that this occurs when the source's parent block and the target block are of the same list type (in this example, ordered lists - but also occurring when they are both unordered lists).

**Comment by ianstormtaylor** [comment]
I believe that this may be fixed by https://github.com/ianstormtaylor/slate/pull/3093, which has changed a *lot* of the logic in Slate and `slate-react` especially. I'm going to close this out, but as always, feel free to open a new issue if it persists for you. Thanks for understanding.

**Comment by jankurianski** [comment]
I retested this bug with slate.js 0.50.8 and can confirm that it has been fixed by #3093  Thank you Ian 👍
